msprime/intervals.py

```diff
@@ -33,83 +33,232 @@ class RateMap:
     """
     A class mapping a numeric value to a set of adjacent intervals along the
     genome.
+
+    :param list position: A list of :math:`n+1` positions, starting at 0, and ending
+        in the sequence length over which the RateMap will apply.
+    :param list rate: A list of :math:`n` positive rates that apply between each
+        position.
+    :param float start_position: When returning a mean_rate over the genome, ignore the
+        part of the map from zero up to ``start_position``, as long as the rate is 0 in
+        that region. If ``None`` (default), the ``start_position`` is 0.
+    :param float end_position: When returning a mean_rate over the genome, ignore the
+        part of the map from ``end_position`` onwards, as long as the rate is 0 in that
+        region. If ``None`` (default), the ``end_position`` is the end of the map.
     """
 
-    def __init__(self, position, rate, map_start=0):
-        # We take copies to make sure there are no unintended consequences
-        # later when a user modifies the arrays in this map.
-        self.position = np.array(position, dtype=float, copy=True)
-        self.rate = np.array(rate, dtype=float, copy=True)
-        self.map_start = map_start
-        size = len(self.rate)
-        if size < 1:
+    def __init__(
+        self,
+        position,
+        rate,
+        *,
+        start_position=None,
+        end_position=None,
+    ):
+        # Taking copies ensures there are no unintended consequences later if a user
```

Comment:
I'm not sure we need to do this any more - we're going to some lengths to make sure the user can't write into the arrays, right? Numpy should manage the read-only view semantics for us - but we should set this flag immediately so this is clear.

Reply:
Good point. Can probably remove

Reply:
Should I leave soemthing like `self._position = np.array(position, dtype=float)` but just remove `copy=True`, which would allow lists or numpy int arrays to be passed in as well as numpy float arrays,?